client/src/details/SFSClientImpl.cpp

```diff
@@ -140,23 +158,26 @@ void ValidateRequestParams(const RequestParams& requestParams, const ReportingHa
 
     for (const auto& [product, _] : requestParams.productRequests)
     {
-        THROW_CODE_IF_LOG(InvalidArg, product.empty(), handler, "product cannot be empty");
+        THROW_CODE_IF_LOG(InvalidArg, product.empty(), handler, "product must not be empty");
     }
 }
 } // namespace
 
 template <typename ConnectionManagerT>
 SFSClientImpl<ConnectionManagerT>::SFSClientImpl(ClientConfig&& config)
-    : m_accountId(std::move(config.accountId))
-    , m_instanceId(config.instanceId && !config.instanceId->empty() ? std::move(*config.instanceId)
-                                                                    : c_defaultInstanceId)
-    , m_nameSpace(config.nameSpace && !config.nameSpace->empty() ? std::move(*config.nameSpace) : c_defaultNameSpace)
 {
     if (config.logCallbackFn)
     {
         m_reportingHandler.SetLoggingCallback(std::move(*config.logCallbackFn));
     }
 
+    ValidateClientConfig(config, m_reportingHandler);
+
+    m_accountId = std::move(config.accountId);
+    m_instanceId =
+        (config.instanceId && !config.instanceId->empty()) ? std::move(*config.instanceId) : c_defaultInstanceId;

```

Comment:
seems like this logic should be in ClientConfig, not here, e.g.

config.GetInstanceId() // returns _instanceId, or if empty returns (static const) ClientConfig::c_defaultInstanceId

Reply:
That makes it harder to move the underlying string from the optional without a getter that returns an rvalue

Reply:
`std::string& GetInstanceId();` and use  `static const std:::string c_defaultInstanceId;` ?